Automattic/jetpack
# tools: Add a tool to help in updating composer dependencies

#### Changes proposed in this Pull Request:

If you make a minor or major change to a package, that's going to bump
the package's branch-alias version number. Which then means you have to
update every project that depends on the package, and any composer.lock
files too.

This adds tools/check-composer-deps.sh to automate that update.

#### Jetpack product discussion


p1616091505148500/1616087978.141100-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






Try each of these on its own new, clean branch. Or clean up everything shown by `git status` in between tests.

* Change the `branch-alias` version in projects/packages/constants/composer.json, then run `tools/check-composer-deps.sh -u`. Does it update everything that depends on that to match, and add change files to match?
* Reset Jetpack's composer.lock to an old version, then run `tools/check-composer-deps.sh -u`. Does it update it and add a changelog entry to match?
* Do both of those at the same time. Make sure it error out trying to create the same-named change file twice for Jetpack.
